new file:   composer.json

AghyadSy opened the pull request:
new file:   composer.lock
	new file:   medicine_warehouse/app/Http/Controllers/Auth/AuthenticatedSessionController.php
	new file:   medicine_warehouse/app/Http/Controllers/Auth/EmailVerificationNotificationController.php
	new file:   medicine_warehouse/app/Http/Controllers/Auth/NewPasswordController.php
	new file:   medicine_warehouse/app/Http/Controllers/Auth/PasswordResetLinkController.php
	new file:   medicine_warehouse/app/Http/Controllers/Auth/RegisteredUserController.php
	new file:   medicine_warehouse/app/Http/Controllers/Auth/VerifyEmailController.php
	modified:   medicine_warehouse/app/Http/Controllers/MedicineController.php
	modified:   medicine_warehouse/app/Http/Kernel.php
	new file:   medicine_warehouse/app/Http/Middleware/EnsureEmailIsVerified.php
	new file:   medicine_warehouse/app/Http/Requests/Auth/LoginRequest.php
	deleted:    medicine_warehouse/app/Http/Resources/MedicineResource.php
	modified:   medicine_warehouse/app/Models/User.php
	modified:   medicine_warehouse/app/Providers/AuthServiceProvider.php
	modified:   medicine_warehouse/app/Providers/RouteServiceProvider.php
	modified:   medicine_warehouse/composer.json
	modified:   medicine_warehouse/composer.lock
	modified:   medicine_warehouse/config/app.php
	modified:   medicine_warehouse/config/cors.php
	modified:   medicine_warehouse/config/sanctum.php
	modified:   medicine_warehouse/database/migrations/2014_10_12_000000_create_users_table.php
	deleted:    medicine_warehouse/package.json
	deleted:    medicine_warehouse/resources/css/app.css
	deleted:    medicine_warehouse/resources/js/app.js
	deleted:    medicine_warehouse/resources/js/bootstrap.js
	new file:   medicine_warehouse/resources/views/.gitkeep
	deleted:    medicine_warehouse/resources/views/welcome.blade.php
	modified:   medicine_warehouse/routes/api.php
	new file:   medicine_warehouse/routes/auth.php
	modified:   medicine_warehouse/routes/web.php
	new file:   medicine_warehouse/tests/Feature/Auth/AuthenticationTest.php
	new file:   medicine_warehouse/tests/Feature/Auth/EmailVerificationTest.php
	new file:   medicine_warehouse/tests/Feature/Auth/PasswordResetTest.php
	new file:   medicine_warehouse/tests/Feature/Auth/RegistrationTest.php
	deleted:    medicine_warehouse/vite.config.js

AghyadSy committed:
new file:   composer.json
	new file:   composer.lock
	new file:   medicine_warehouse/app/Http/Controllers/Auth/AuthenticatedSessionController.php
	new file:   medicine_warehouse/app/Http/Controllers/Auth/EmailVerificationNotificationController.php
	new file:   medicine_warehouse/app/Http/Controllers/Auth/NewPasswordController.php
	new file:   medicine_warehouse/app/Http/Controllers/Auth/PasswordResetLinkController.php
	new file:   medicine_warehouse/app/Http/Controllers/Auth/RegisteredUserController.php
	new file:   medicine_warehouse/app/Http/Controllers/Auth/VerifyEmailController.php
	modified:   medicine_warehouse/app/Http/Controllers/MedicineController.php
	modified:   medicine_warehouse/app/Http/Kernel.php
	new file:   medicine_warehouse/app/Http/Middleware/EnsureEmailIsVerified.php
	new file:   medicine_warehouse/app/Http/Requests/Auth/LoginRequest.php
	deleted:    medicine_warehouse/app/Http/Resources/MedicineResource.php
	modified:   medicine_warehouse/app/Models/User.php
	modified:   medicine_warehouse/app/Providers/AuthServiceProvider.php
	modified:   medicine_warehouse/app/Providers/RouteServiceProvider.php
	modified:   medicine_warehouse/composer.json
	modified:   medicine_warehouse/composer.lock
	modified:   medicine_warehouse/config/app.php
	modified:   medicine_warehouse/config/cors.php
	modified:   medicine_warehouse/config/sanctum.php
	modified:   medicine_warehouse/database/migrations/2014_10_12_000000_create_users_table.php
	deleted:    medicine_warehouse/package.json
	deleted:    medicine_warehouse/resources/css/app.css
	deleted:    medicine_warehouse/resources/js/app.js
	deleted:    medicine_warehouse/resources/js/bootstrap.js
	new file:   medicine_warehouse/resources/views/.gitkeep
	deleted:    medicine_warehouse/resources/views/welcome.blade.php
	modified:   medicine_warehouse/routes/api.php
	new file:   medicine_warehouse/routes/auth.php
	modified:   medicine_warehouse/routes/web.php
	new file:   medicine_warehouse/tests/Feature/Auth/AuthenticationTest.php
	new file:   medicine_warehouse/tests/Feature/Auth/EmailVerificationTest.php
	new file:   medicine_warehouse/tests/Feature/Auth/PasswordResetTest.php
	new file:   medicine_warehouse/tests/Feature/Auth/RegistrationTest.php
	deleted:    medicine_warehouse/vite.config.js

Amajd031 commented on medicine_warehouse/app/Http/Controllers/Auth/NewPasswordController.php:
```diff
@@ -0,0 +1,53 @@
+    {
+        $request->validate([
+            'token' => ['required'],
+            'email' => ['required', 'email'],
```
Make it a phone

Amajd031 commented on medicine_warehouse/app/Http/Controllers/Auth/PasswordResetLinkController.php:
```diff
@@ -0,0 +1,39 @@
+    public function store(Request $request): JsonResponse
+    {
+        $request->validate([
+            'email' => ['required', 'email'],
```
phone...

Amajd031 commented on medicine_warehouse/app/Http/Controllers/Auth/PasswordResetLinkController.php:
```diff
@@ -0,0 +1,39 @@
+        // to send the link, we will examine the response then see the message we
+        // need to show to the user. Finally, we'll send out a proper response.
+        $status = Password::sendResetLink(
+            $request->only('email')
```
phone

Amajd031 commented on medicine_warehouse/app/Http/Requests/Auth/LoginRequest.php:
```diff
@@ -0,0 +1,99 @@
+        throw ValidationException::withMessages([
+            'email' => trans('auth.throttle', [
+                'seconds' => $seconds,
+                'minutes' => ceil($seconds / 60),
```
add a phone number function

Amajd031 commented on medicine_warehouse/database/migrations/2014_10_12_000000_create_users_table.php:
```diff
@@ -15,9 +15,9 @@ public function up(): void
             $table->id();
             $table->string('name');
             $table->string('email')->unique();
+            $table->string('phone');
```
make uniqe

Amajd031 commented on medicine_warehouse/tests/Feature/Auth/AuthenticationTest.php:
```diff
@@ -0,0 +1,47 @@
+
+        $response = $this->post('/login', [
+            'email' => $user->email,
+            'password' => 'password',
```
phone...

Amajd031 commented on medicine_warehouse/tests/Feature/Auth/AuthenticationTest.php:
```diff
@@ -0,0 +1,47 @@
+
+        $this->post('/login', [
+            'email' => $user->email,
+            'password' => 'wrong-password',
```
phone...

Amajd031 commented on medicine_warehouse/tests/Feature/Auth/PasswordResetTest.php:
```diff
@@ -0,0 +1,47 @@
+            return true;
+        });
+    }
+}
```
We are not going to use these features

Amajd031 commented on medicine_warehouse/tests/Feature/Auth/RegistrationTest.php:
```diff
@@ -0,0 +1,24 @@
+    {
+        $response = $this->post('/register', [
+            'name' => 'Test User',
+            'email' => 'test@example.com',
```
phone..

AghyadSy committed:
modified:   medicine_warehouse/app/Http/Controllers/Auth/RegisteredUserController.php
	modified:   medicine_warehouse/app/Http/Controllers/ClassificationController.php
	modified:   medicine_warehouse/app/Http/Controllers/CompanyOfMedicineController.php
	modified:   medicine_warehouse/app/Http/Controllers/MedicineController.php
	modified:   medicine_warehouse/app/Http/Controllers/MedicineOrderController.php
	modified:   medicine_warehouse/app/Http/Controllers/WarehouseController.php
	modified:   medicine_warehouse/app/Http/Requests/Auth/LoginRequest.php
	modified:   medicine_warehouse/app/Models/Medicine.php
	modified:   medicine_warehouse/app/Models/MedicineOrder.php
	modified:   medicine_warehouse/app/Models/User.php
	modified:   medicine_warehouse/database/migrations/2014_10_12_000000_create_users_table.php
	modified:   medicine_warehouse/routes/auth.php

AghyadSy committed:
modified:   medicine_warehouse/app/Http/Controllers/ClassificationController.php
	modified:   medicine_warehouse/app/Http/Controllers/CompanyOfMedicineController.php
	modified:   medicine_warehouse/app/Http/Controllers/MedicineController.php
	modified:   medicine_warehouse/app/Http/Controllers/MedicineOrderController.php
	modified:   medicine_warehouse/app/Http/Controllers/PharmaciyController.php
	modified:   medicine_warehouse/app/Http/Controllers/WarehouseController.php
